boards: longan_nano: add dfu-util runner

ottojo opened the pull request:
This adds the runner arguments for dfu-util to the longan_nano board, to allow flashing via DFU in addition to openocd.

I added the `CONFIG_FLASH_BASE_ADDRESS` since that results in the proper `dfu-util` flag `-s 0x08000000`, which i found in the [platformio sources from sipeed](https://github.com/sipeed/platform-gd32v/blob/master/builder/main.py#L146).
I'm not sure if this has other, unintended effects... (This option also did not seem available for RISCV, was this on purpose?)

ottojo committed:
arch: riscv enable flash config

For RISCV arch, enable FLASH_SIZE and FLASH_BASE_ADDRESS config.
To avoid duplicated work, remove flash config from RISCV soc.

Signed-off-by: Jonas Otto <jonas@jonasotto.com>

cameled commented on boards/riscv/longan_nano/longan_nano_defconfig:
```diff
@@ -10,6 +10,8 @@ CONFIG_BOARD_LONGAN_NANO=y
 
 CONFIG_GD32_HXTAL_8MHZ=y
 
+CONFIG_FLASH_BASE_ADDRESS=0x08000000
```
Normally it auto detected through `dt_chosen_reg_addr_hex`. I add PR #54967 to fix this issue.

ottojo replied:
do i understand correctly that i should cherry-pick 95da6b56da85ec23443f6d28662f7f4159210846, and divide this MR into separate commits (adding the flash-config to longan_nano, and adding the dfu settings)?

soburi replied:
This file may no need to change because the CONFIG_FLASH_BASE be set correctly by `arch/Kconfig` settings that enabled by this PR. (Please check generated .config file.)

soburi replied:
This file may no need to change because the CONFIG_FLASH_BASE be set correctly by arch/Kconfig settings that enabled by this PR. (Please check generated .config file.)

soburi replied:
Off topic:
Reviewers reviews a lot of PRs, and they press this button to indicate that their issue has been resolved.
It is better not to press this button yourself.
(People forget easy what say even yesterday. The `Resolve conversation` button works as reviewer's reminder.)

cameled replied:
Hi @ottojo , No need to cherry-pick, you already make very well works. It just my suggestion.

ottojo commented on boards/riscv/longan_nano/board.cmake:
```diff
@@ -8,3 +8,7 @@ board_runner_args(openocd "--cmd-load=gd32vf103-load")
 
 include(${ZEPHYR_BASE}/boards/common/openocd.board.cmake)
+
+board_runner_args(dfu-util "--pid=28e9:0189" "--alt=0" "--dfuse")
```
i grouped the new `board_runner_args` with the existing ones above :+1:

soburi replied:
Off topic:
Reviewers reviews a lot of PRs, and they press this button to indicate that their issue has been resolved.
It is better not to press this button yourself.
(People forget easy what say even yesterday. The `Resolve conversation` button works as reviewer's reminder.)

ottojo committed:
boards: longan_nano: add dfu-util runner

This adds the runner arguments for dfu-util to the longan_nano board,
to allow flashing via DFU in addition to openocd.

Signed-off-by: Jonas Otto <jonas@jonasotto.com>

ottojo commented:
(sorry for removing the review-requests? i think manually requesting re-review interfered with the bot somehow...)

cameled commented on soc/riscv/riscv-privilege/gd32vf103/Kconfig.defconfig.gd32vf103:
```diff
@@ -34,9 +34,6 @@ config NUM_IRQS
 	default 87 if  NUCLEI_ECLIC
 	default 16 if !NUCLEI_ECLIC
 
-config FLASH_BASE_ADDRESS
```
Since you removed this, The CONFIG_FLASH_BASE_ADDRESS will be detected correctly.

cameled commented on soc/riscv/esp32c3/Kconfig.soc:
```diff
@@ -26,14 +26,6 @@ config ESPTOOLPY_FLASHFREQ_80M
 
-config FLASH_SIZE
-	int
-	default $(dt_node_reg_size_int,/soc/flash-controller@60002000/flash@0,0)
```
There have some CI failure caused by esp32c3 not set `unit`, but `FLASH_SIZE` in `arch/Kconfig` set `uint` to `K`.
```
arch/Kconfig:
  default $(dt_chosen_reg_size_int,$(DT_CHOSEN_Z_FLASH),0,K) if (XIP && (ARM ||ARM64)) || !ARM

esp32c3/Kconfig.soc:
  default $(dt_node_reg_size_int,/soc/flash-controller@60002000/flash@0,0)
```
I think we need to keep the `FLASH_SIZE` and `FLASH_BASE_ADDRESS` config for esp32c3.

ottojo replied:
done!

sylvioalves replied:
@cameled @ottojo, thanks for keeping ESP32-C3 as is, otherwise it would not work.

cameled commented:
Hi @ottojo , any update?

ottojo commented:
> This file may no need to change because the CONFIG_FLASH_BASE be set correctly by arch/Kconfig settings that enabled by this PR. (Please check generated .config file.)
@soburi yes this is the case, i removed the explicit CONFIG_FLASH_BASE setting

@cameled i re-added the configuration in `esp32c3/Kconfig.soc`

cameled reviewed (changes requested):
Hi @ottojo, thanks for you contirbutions! One thing left now, please update the `signed-off-by` info in `arch` commit. Remove my info and add your's.

```
$ git rebase -i c626078921d515b283ec40823deb52bce1cccafb^

$ git commit --amend -s
```

ottojo commented:
> Hi @ottojo, thanks for you contirbutions! One thing left now, please update the `signed-off-by` info in `arch` commit. Remove my info and add your's.

done!
